Improve the SDS doc. (#3139)
istio/istio.io: content/docs/tasks/security/auth-sds/index.md

  which causes performance regression.

* Potential security vulnerability:
  The workload private keys are distributed through Kubernetes secrets, which have known
  [risks](https://kubernetes.io/docs/concepts/configuration/secret/#risks). Also, the private key is stored in the pod local file system.

These issues are addressed in Istio 1.1 through the SDS identity provision flow. The workflow can be described as follows.

The workload sidecar Envoy requests key and certificates from the Citadel Agent (SDS server, which runs as per-node `DaemonSet`) using a Kubernetes
service account JWT through the SDS API.
The Citadel Agent generates a key pair and sends the CSR request to Citadel. Citadel verifies the JWT and issues the certificate to the Citadel Agent.
Eventually, Citadel Agent sends the key and certificate back to the workload sidecar.
In this approach, the private key never leaves the node; it is only kept in the Citadel Agent and Envoy sidecar's memory.
Thus, this approach is more secure than the secret volume mount approach, which relies on the secret and local files on pod.

> Also, the sidecar Envoy is able to dynamically renew the key and certificate through the SDS API, so restarting Envoy can be avoided.

Reviewer: The list also helps make the actors and actions a lot clearer:
```suggestion
* The sidecar Envoy can dynamically renew the key and certificate through the SDS API: Certificate rotations no longer require Envoy to restart.
```

sleep.bar to httpbin.bar: 200
{{< /text >}}

## Verifying no secret-volume mounted file is generated

> Verify no secret-volume mounted file is generated by remotely accessing to workload sidecar container that we just deployed:

Reviewer: The text became convoluted and difficult to follow with the added adjective. To simplify the language to make the text more understandable I suggest:
```suggestion
To verify that no secret-volume mounted file is generated, remotely access the deployed workload sidecar container:
```

$ kubectl delete ns bar
$ kubectl delete -f istio-auth-sds.yaml
{{< /text >}}

## Caveats

Currently, the SDS identity provision flow has the following caveats:

> * The control plane security has to be turned off. Enabling SDS with control plane security turned on is working in progress.

Reviewer: The text fails to communicate how the one impacts the other. Make that impact explicit for clarity:
```suggestion
* To enable SDS, you must turn off the control plane security. Enabling SDS while the control plane security is on remains a work in progress.
```

> * The support clusters smoothly migrating from using the secret volume mount to using the SDS is working in progress.

Reviewer: What's the caveat here? I'm guessing that the migration is not smooth but the text does not clearly convey that idea. 
```suggestion
* There is no smooth migration for support clusters to stop using the secret volume mount approach. A smooth migration for the support clusters to use the SDS approach is a work in progress.
```
